charts/external-dns/templates/deployment.yaml

```
@@ -54,6 +54,10 @@ spec:
       {{- with .Values.dnsPolicy }}
       dnsPolicy: {{ . }}
       {{- end }}
+      {{- with .Values.initContainers }}
+      initContainers:
+        {{- toYaml . | nindent 8 }}
```

```suggestion
        {{- tpl (toYaml .) . | nindent 8 }}
```
By using `tpl`, it will allow to use `{{ include "external-dns.image" . }}` in the values

---

@mloiseleur that would be incorrect, `tpl` shouldn't be used like this. If you want `tpl` support you'd need to support init containers as either a string or an array and only template when it's been provided as a string. I'd prefer this to stay as it is.

---

@calvinbui It has been done on Traefik Helm Chart, see [this PR](https://github.com/traefik/traefik-helm-chart/pull/557) for instance.
@stevehipwell I understand. It's ok for me.

---

@mloiseleur this is a well known anti-pattern that looks like it should work, works in very simple scenarios but fails hard for anything more complex. To prove this attempt to use the selector labels template for traefik affinity match labels and you will see what I mean.

---

Yes, you're right.